### ISSUE TITLE
LocalNotifications.addOnMessageReceivedCallback not called if app was stopped

### ISSUE THREAD

**Issue by memphisvl** [EddyVerbruggen/nativescript-local-notifications#53]
I've been trying to figure out this for quite some time. The plugin works very well, but in this scenario I can't seem to make it work.

1. Schedule a local notification
2. Stop the app (remove from running in Android/iOS)
3. Notification appears in the notification bar (both Android & iOS)
4. Click on notification starts the app
5. LocalNotifications.addOnMessageReceivedCallback is not called

**Versions:**
{N} 2.5.0
Plugin: 1.2.1

```javascript
LocalNotifications.addOnMessageReceivedCallback(data => {
  alert('Id: '+data.id+' Title: '+data.title);
})

// Do I need to register it before or after is ok as well?
platformNativeScriptDynamic().bootstrapModule(AppModule);
```

The scenario I have requires to get notification data in order to show other screen then home.

**Comment by mikraus94** [comment]
I have same issue, but only on Android.  On iOS works correctly.

**Comment by ickata** [comment]
Hi all,

On my end this is working pretty well. Make sure that you register your onMessageReceived callback as close to the top of your app as possible. If you do this in Page onload for example, or inside some async callback – that might be the cause.

**Comment by mikraus94** [comment]
I've registred onMessageReceived by adding local notifications service to providers in app modules and it doesn't work. Have you better idea how to solve it?

**Comment by mikraus94** [comment]
Okay, Ive' made change to code. Now, Android works but iOS doesn't work.

**Comment by ickata** [comment]
Would you please tell what change you've made?

**Comment by mikraus94** [comment]
I've made init method with addOnMessageReceivedCallback in local notification service and call it in first component (app.component.ts)

**Comment by Ericky14** [comment]
Are there any updated on this? It works fine on the iOS emulator, but on a real iOS device the receive callback is never called when the app is suspended. But when the app is resumed, the callback from when it was suspended is called.

**Comment by Lacos247** [comment]
Same issue here. Tested in Combination with the other firebase plugin Version 5.0.2 and 4.1.2..

**Comment by EddyVerbruggen** [comment]
This may be fixed in plugin version 3.0.0 because notifications now use a the new iOS 10 `UNNotificationCenter` framework and the corresponding delegate methods, which seem a little more background-friendly.

**Comment by StefanNedelchev** [comment]
> This may be fixed in plugin version 3.0.0 because notifications now use a the new iOS 10 `UNNotificationCenter` framework and the corresponding delegate methods, which seem a little more background-friendly.

It might be fixed on iOS but it's still broken on Android.
